### PR TITLE
Fix a couple of warnings found by PVS Studio

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
# Issue

I asked for a [PVS Studio](https://pvs-studio.com/en/) license for us(they give it for free for open source projects) and it was able to find a couple of issues like this.

## Tasklist

 - [ ] CHANGELOG.md entry ([How to write a changelog entry](http://keepachangelog.com/en/1.0.0/#how))
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md))
 - [ ] review
 - [ ] adjust for comments
 - [ ] cherry pick to release branch

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: include/util/coordinate_calculation.hpp [hunk 1]
```diff
@@ -372,7 +372,7 @@ bool areParallel(const iterator_type lhs_begin,
     const auto slope_rhs = get_slope(null_island, rotated_difference_rhs);
     // the left hand side has a slope of `0` after the rotation. We can check the slope of the right
     // hand side to ensure we only considering slight slopes
-    return std::abs(slope_rhs) < 0.20; // twenty percent incline at the most
+    return std::fabs(slope_rhs) < 0.20; // twenty percent incline at the most
```
Comment (SiarheiFedartsou) [inline include/util/coordinate_calculation.hpp:375]: This one is the most interesting - std::abs has overloads only for integer types. 
https://en.cppreference.com/w/cpp/numeric/math/abs

File: .github/workflows/osrm-backend.yml [hunk 2]
```diff
@@ -607,7 +607,6 @@ jobs:
       
         # Linux dev packages
         if [ "${TARGET_ARCH}" != "i686" ] && [ "${ENABLE_CONAN}" != "ON" ]; then
-          sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
```
Comment (SiarheiFedartsou) [inline .github/workflows/osrm-backend.yml:610]: It started failing unexpectedly, but it turned out that we probably don't need it :)

**Comment by SiarheiFedartsou** [comment]
@mjjbell please take a look when you have a chance :)

File: .github/workflows/osrm-backend.yml (hunk 2 above)
Comment (mjjbell) [reply .github/workflows/osrm-backend.yml:610]: Yes I think we're lucky that 20.04 supports the 5 gcc versions in our matrix. If the distro and matrix get out of sync, we'll need to bring it back (or just limit it to the versions the distro supports).

**Review by mjjbell — APPROVED** [review]
Is this something we can run continuously?

I see they have a docs page for running via Github Actions: https://pvs-studio.com/en/docs/manual/6579/

**Comment by SiarheiFedartsou** [comment]
> Is this something we can run continuously?
> 
> I see they have a docs page for running via Github Actions: https://pvs-studio.com/en/docs/manual/6579/

We definitely can, will try to look into it at some point. 

Btw will you be able to take a look into other opened PRs? you probably missed notifications about them(or just don’t have time for it, not sure if you have enough time at the moment)

**Pull request merged**